src/index.tsx

```diff
@@ -311,6 +310,7 @@ function SwipeableItem<T>(
             : closestSnapPoint > 0
             ? OpenDirection.RIGHT
             : OpenDirection.LEFT;
+        runOnJS(setOpenDirection)(openDirection)
```

Comment:
strange that this fixes the issue, any idea why?
I would expect this to be exactly the same as calling `setOpenDirection` on the first line in `onAnimationEnd`

What about all the other cases that `onAnimationEnd` is called? Wouldn't removing `setOpenDirection` cause incorrect state in those cases?

Reply:
So the issue are seeing is that the underlay is not intractable until after the animation is finished so if the user tries to press a button in the underlay while animation is "easing out" it wont work because pointer events are disabled [here](https://github.com/computerjazz/react-native-swipeable-item/blob/main/src/index.tsx#L385) and this value isn't updated until the animation is completely finished.  So the fix here is the underlay being intractable when the animation starts and this accomplishes that by setting the open direction sooner than it did before.   I'm not sure about downstream issues from this but wanted to chime in with context as to what this is solving / why it solves.

Reply:
can you try `2.0.8` and see if it fixes this issue?